### ISSUE TITLE
Using gr.Markdown as output of gr.Button disables the loading spinner

### ISSUE THREAD

**Issue by maxkskhor** [gradio-app/gradio#10165]
### Describe the bug

Using `gr.Markdown` as output of `gr.Button` disables the loading spinner.

The spinner works in other components such as `gr.Textbox` or `gr.HTML`.

The loading spinner worked in **Gradio 4.42.0** but doesn't work in **Gradio 5.8.0**.

The user would not know if the app is running as the loading spinner is not working for `gr.Markdown`.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
import gradio as gr
import time


def func(x):
    time.sleep(5)
    return x


with gr.Blocks() as demo:
    text = gr.Textbox()
    # output = gr.Textbox()
    output = gr.Markdown()

    run = gr.Button('Run')
    run.click(
        fn=func,
        inputs=[text],
        outputs=[output]
    )


if __name__ == '__main__':
    demo.launch()
  

```

### Screenshot

![image](https://github.com/user-attachments/assets/2a138bcf-2b85-42e4-8a67-3746df503558)

### Logs

```shell
No logs. It is a UI issue.
```

### System Info

```shell
Gradio Environment Information:
------------------------------
Operating System: Windows
gradio version: 5.8.0
gradio_client version: 1.5.1

------------------------------------------------
gradio dependencies in your environment:

aiofiles: 23.2.1
anyio: 4.7.0
audioop-lts is not installed.
fastapi: 0.115.6
ffmpy: 0.4.0
gradio-client==1.5.1 is not installed.
httpx: 0.28.1
huggingface-hub: 0.26.5
jinja2: 3.1.4
markupsafe: 2.1.5
numpy: 2.2.0
orjson: 3.10.12
packaging: 24.2
pandas: 2.2.3
pillow: 10.4.0
pydantic: 2.10.3
pydub: 0.25.1
python-multipart: 0.0.19
pyyaml: 6.0.2
ruff: 0.8.2
safehttpx: 0.1.6
semantic-version: 2.10.0
starlette: 0.41.3
tomlkit: 0.12.0
typer: 0.15.1
typing-extensions: 4.12.2
urllib3: 2.2.3
uvicorn: 0.32.1
authlib; extra == 'oauth' is not installed.
itsdangerous; extra == 'oauth' is not installed.


gradio_client dependencies in your environment:

fsspec: 2024.10.0
httpx: 0.28.1
huggingface-hub: 0.26.5
packaging: 24.2
typing-extensions: 4.12.2
websockets: 12.0
```

### Severity

I can work around it.

**Comment by maxkskhor** [comment]
Found the root issue - the loading spinner is disabled only when the `value=None` (which is the default value). Set the value to an empty string `value=''` would make the loading spinner work again.

Similar bug to this issue: https://github.com/gradio-app/gradio/issues/10162.

Defaulting to `value=None` causes weird UI issues. I encountered these issues a lot since I often use `visible=False` to render a component later and just let value defaulted to `None`.
